From 503812c33d9049ded3d91bbea0e66ab8ea873d4e Mon Sep 17 00:00:00 2001
From: Florinel Iordache <florinel.iordache@nxp.com>
Date: Mon, 5 Nov 2018 17:04:05 +0200
Subject: [PATCH 034/741] arm64: dts: Added endianness information to dts
 serdes node

commit 132c27081d33cc110692f4809be28bb0cb562f72 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

This change is required to specify that serdes HW peripheral is little-endian.

Signed-off-by: Florinel Iordache <florinel.iordache@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-ls1088a.dtsi | 3 ++-
 arch/arm64/boot/dts/freescale/fsl-ls208xa.dtsi | 3 ++-
 arch/arm64/boot/dts/freescale/fsl-lx2160a.dtsi | 3 ++-
 3 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1088a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1088a.dtsi
index 31718f9f1f06..2fab967454ad 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1088a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1088a.dtsi
@@ -891,8 +891,9 @@
                 };
 
                 serdes1: serdes@1ea0000 {
-                                reg = <0x0 0x1ea0000 0 0x00002000>;
                                 compatible = "fsl,serdes-10g";
+				reg = <0x0 0x1ea0000 0 0x00002000>;
+				little-endian;
                 };
 	};
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls208xa.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls208xa.dtsi
index 99a93b8ca51f..6dc8c8a51d4b 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls208xa.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls208xa.dtsi
@@ -825,8 +825,9 @@
 		};
 
 		serdes1: serdes@1ea0000 {
-			reg = <0x0 0x1ea0000 0 0x00002000>;
 			compatible = "fsl,serdes-10g";
+			reg = <0x0 0x1ea0000 0 0x00002000>;
+			little-endian;
 		};
 
 		ccn@4000000 {
diff --git a/arch/arm64/boot/dts/freescale/fsl-lx2160a.dtsi b/arch/arm64/boot/dts/freescale/fsl-lx2160a.dtsi
index 7b0639a3f422..b93bae2bc054 100644
--- a/arch/arm64/boot/dts/freescale/fsl-lx2160a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-lx2160a.dtsi
@@ -581,8 +581,9 @@
 		};
 
 		serdes1: serdes@1ea0000 {
-				reg = <0x0 0x1ea0000 0 0x00002000>;
 				compatible = "fsl,serdes-28g";
+				reg = <0x0 0x1ea0000 0 0x00002000>;
+				little-endian;
 		};
 
 		i2c0: i2c@2000000 {
-- 
2.17.1

